tests/random_test.py

```diff
@@ -1853,6 +1853,24 @@ def test_device_put_replicated(self):
     keys_on_device = jax.device_put_replicated(key, devices)
     self.assertArraysEqual(jnp.broadcast_to(key, keys_on_device.shape), keys_on_device)
 
+  def test_make_array_from_callback(self):
+    devices = jax.devices()
+    shape = (len(devices),) if config.jax_enable_custom_prng else (len(devices), 2)
+    sharding = jax.sharding.PmapSharding.default(shape, devices=devices)
+    callback = lambda index: jax.random.PRNGKey(index[0])
+    result = jax.make_array_from_callback(shape, sharding, callback)
+    expected = jax.vmap(jax.random.PRNGKey)(jnp.arange(len(devices)))
+    self.assertArraysEqual(result, expected)
+
+  def test_make_array_from_single_device_arrays(self):
+    devices = jax.devices()
+    shape = (len(devices),) if config.jax_enable_custom_prng else (len(devices), 2)
+    sharding = jax.sharding.PmapSharding.default(shape, devices=devices)
```

Comment:
Maybe use NamedSharding instead of PmapSharding?

NamedSharding is more important nowadays ;)

Reply:
Sure, how would you construct a default `NamedSharding` here?

Reply:
mesh = jtu.create_global_mesh((2, 2), ('x', 'y'))
s = jax.sharding.NamedSharding(mesh, P('x', 'y'))

Change the pspec depending on your input. If it is a scalar use `P()`.

Reply:
Ah, so I'll need to branch both the mesh and the sharding definitions on the value of `jax_enable_custom_prng`, because the shape changes...

Seems simpler to just stick with PmapSharding, is there any reason not to?

Reply:
Because PmapSharding will go away :) (and I hate it)

> sharding definitions on the value of jax_enable_custom_prng, because the shape changes
Sounds fine? We do that in a bunch of places. You'll just have to change the pspec. mesh and other stuff stays the same.

Reply:
After all, this is basically a smoke test to make sure we don't get an error, not a comprehensive test of sharding dynamics

Reply:
To be clear, my understanding is I'd have to replace this:
```python
sharding = jax.sharding.PmapSharding.default(shape, devices=devices)
```
with this:
```python
if jax.config.jax_enable_custom_prng:
  mesh = jtu.create_global_mesh((len(devices),), ('x',))
  s = jax.sharding.NamedSharding(mesh, P())
else:
  mesh = jtu.create_global_mesh((len(devices), 2), ('x', 'y'))
  s = jax.sharding.NamedSharding(mesh, P('x', 'y'))
```
Not the end of the world, but seems like a lot of complexity to add in a test if it's not absolutely necessary.

Reply:
Yeah, but NamedSharding will get better real world use cases so it's better to test that. I mean you should test both but if you are going to choose one, then go with namedSharding

Reply:
Wait.. you can just keep the thing in the if branch and that will work across anything.

Reply:
OK I tried it... with `NamedSharding` the `callback` no longer gets the row index as an argument. Why is that?

Reply:
I also tried just keeping the `True` branch – this results in an incorrect output shape when `jax_custom_prng_key=False`, because we're sharding across the key dimension rather than the batch dimension.

Reply:
Alright, changed to `NamedSharding` – thanks for the review!